src/intrinsics/ecma262/Object.js

```diff
@@ -140,7 +143,15 @@ export default function(realm: Realm): NativeFunctionValue {
     }
 
     // 5. Return to.
-    if (to_must_be_partial) to.makePartial();
+    if (to_must_be_partial) {
+      to.makePartial();
+      let to_keys = to.$OwnPropertyKeys();
```

Comment:
Shouldn’t this run before we assigned all the properties? By now it already has keys (we just set them, no?)

If this observation is correct then I’m surprised how the tests could pass. Is test stressing this code path at all?

Reply:
I can change that, but it doesn't make any difference for these tests as there are no keys to copy over. I'll add another test that does have keys.